server/schedulers/hot_region.go

```diff
@@ -215,7 +214,7 @@ func (h *balanceHotRegionsScheduler) balanceHotWriteRegions(cluster opt.Cluster)
 				schedulerCounter.WithLabelValues(h.GetName(), "move-peer").Inc()
 				return []*operator.Operator{op}
 			}
-		case 1:
+		} else if h.allowBalanceLeader(cluster) {
```

Comment:
I think it duplicates with `allowBalanceLeader` inside `balanceByLeader`?

Reply:
Yes it is duplicated. But there other places calling `balanceByLeader` too.